File: app/src/main/java/dev/bee/kanjianki/StudyCueTexts.java
package dev.bee.kanjianki;

import dev.bee.kanjianki.core.RecordsImportModels;
import dev.bee.kanjianki.core.RecordsSchedulerModels;
import dev.bee.kanjianki.core.DictionaryLookup;
import dev.bee.kanjianki.core.StudyCuePolicy;

import java.util.List;

final class StudyCueTexts {
    private StudyCueTexts() {
    }

    static List<String> answerLines(
            DictionaryLookup dictionaryLookup,
            RecordsSchedulerModels.StudySession session,
            RecordsImportModels.Example example,
            boolean wordReadingTask
    ) {
        return StudyCuePolicy.answerLines(dictionaryLookup, session, example, wordReadingTask);
    }

    static String displayGlosses(List<String> meanings, int maxMeanings) {
        return StudyCuePolicy.displayGlosses(meanings, maxMeanings);
    }

    static String cleanFallbackMeaning(String raw, String fallback, int maxChars) {
        return StudyCuePolicy.cleanFallbackMeaning(raw, fallback, maxChars);
    }
}

File: app/src/main/kotlin/dev/bee/kanjianki/StudyCueTexts.kt
package dev.bee.kanjianki

import dev.bee.kanjianki.core.DictionaryLookup
import dev.bee.kanjianki.core.RecordsImportModels
import dev.bee.kanjianki.core.RecordsSchedulerModels
import dev.bee.kanjianki.core.StudyCuePolicy

internal object StudyCueTexts {
    @JvmStatic
    fun answerLines(
        dictionaryLookup: DictionaryLookup,
        session: RecordsSchedulerModels.StudySession?,
        example: RecordsImportModels.Example?,
        wordReadingTask: Boolean,
    ): List<String> {
        return StudyCuePolicy.answerLines(dictionaryLookup, session, example, wordReadingTask)
    }

    @JvmStatic
    fun displayGlosses(meanings: List<String>?, maxMeanings: Int): String {
        return StudyCuePolicy.displayGlosses(meanings, maxMeanings)
    }

    @JvmStatic
    fun cleanFallbackMeaning(raw: String?, fallback: String?, maxChars: Int): String {
        return StudyCuePolicy.cleanFallbackMeaning(raw, fallback, maxChars)
    }
}
